src/components/NewCommentForm.tsx

```diff
@@ -1,6 +1,122 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { addNewComment } from '../services/comment';
+import classNames from 'classnames';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+  setError: (value: boolean) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setError,
+}) => {
+  const [newComment, setNewComment] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleOnChange(
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) {
+    if (event.target.name === 'name') {
+      setNewComment(prev => {
+        return {
+          ...prev,
+          name: event.target.value,
+        };
+      });
```

Comment:
```suggestion
      setNewComment(prev => {
        return {
          ...prev,
          [event.target.name]: event.target.value,
        };
      });
```
you can simplify that